ocm/gardener.py

```diff
@@ -0,0 +1,170 @@
+'''
+this module contains code specific to how Gardener-Components (https://github.com/gardener)
+define cross-references to resources to other (Gardener-)Components.
+
+Code in this module is not intended for re-use.
+'''
+
+import collections.abc
+import copy
+import os
+
+import yaml
+
+import oci.model
+import ocm
+
+
+def find_imagevector_file(
+    repo_root: str=None,
+) -> str | None:
+    for candidate in (
+        'charts/images.yaml',
+        'imagevector/images.yaml',
+        'imagevector/container.yaml',
+    ):
+        if repo_root:
+            candidate = os.path.join(
+                repo_root,
+                candidate,
+            )
+        if os.path.isfile(candidate):
+            return candidate
+
+
+def iter_images_from_imagevector(
+    images_yaml_path: str,
+) -> collections.abc.Generator[dict, None, None]:
+    with open(images_yaml_path) as f:
+      for part in yaml.safe_load_all(f):
+        yield from part['images']
+
+
+def add_resources_from_imagevector(
+    component: ocm.Component,
+    image_dicts: collections.abc.Iterable[dict],
+    component_prefixes: list[str],
+) -> ocm.Component:
+  imagevector_label_name = 'imagevector.gardener.cloud/images'
+  imagevector_label = component.find_label(imagevector_label_name)
+  if not imagevector_label:
+    component.labels.append(
+      imagevector_label := ocm.Label(
+        name=imagevector_label_name,
+        value={'images': []}
+      )
+    )
+
+  for image_dict in image_dicts:
+    name = image_dict['name']
+    resource_id = image_dict.get('resourceId', {'name': name})
+    source_repo = image_dict.get('sourceRepository', None)
+    img_repo = image_dict['repository']
+    extra_identity = image_dict.get('extraIdentity', {})
+    labels = copy.copy(image_dict.get('labels', []))
+    tag = image_dict.get('tag', None)
+    target_version = image_dict.get('targetVersion', None)
```

Comment:
Why not introduce a dataclass to describe `image_dict`?
Especially useful considering the default values.

Reply:
we might do that in a separate change. However:

- this pullrequest is foremostly focussed around re-using the existing implementation also for base-component-descriptor-action
- the contract is defined on gardener-side, and is not fully known to us / not explicitly defined (to my knowledge). consuming as raw dict makes this slightly more obvious